src/NLog.Raven/RavenTarget.cs

```diff
@@ -122,11 +128,24 @@ protected override void Write(IList<AsyncLogEventInfo> logEvents)
             {
                 using (var bulkInsert = _documentStore.BulkInsert())
                 {
-                    for (int i = 0; i < logEvents.Count; ++i)
+                    var expiry = DateTime.UtcNow.AddDays(ExpiryDays);
```

Comment:
Consider doing this:

var expiryMetaData = ExpiryDays > 0 ? new MetadataAsDictionary
{
     new KeyValuePair<string, object>(Constants.Documents.Metadata.Expires, DateTime.UtcNow.AddDays(ExpiryDays))
} : null;
```
Then use it like this:
```c#
if (expiryMetaData != null
{
    bulkInsert.Store(logEntry, expirationMetadata);
}
else
{
    bulkInsert.Store(logEntry);
}
```